packages/jsii-rosetta/lib/jsii/assemblies.ts

```diff
@@ -171,36 +186,50 @@ export function allSnippetSources(assembly: spec.Assembly): AssemblySnippetSourc
   }
 }
 
-export function allTypeScriptSnippets(assemblies: readonly LoadedAssembly[], loose = false): TypeScriptSnippet[] {
-  const ret = new Array<TypeScriptSnippet>();
-
-  for (const { assembly, directory } of assemblies) {
-    const strict = enforcesStrictMode(assembly);
-    for (const source of allSnippetSources(assembly)) {
-      switch (source.type) {
-        case 'example':
-          // If an example is an infused example, we do not care about compiler errors.
-          // We are relying on the tablet cache to have this example stored already.
-          const [strictForExample, looseForExample] =
-            source.metadata?.infused !== undefined ? [false, true] : [strict, loose];
-          const location = { api: source.location, field: { field: 'example' } } as const;
-          const snippet = updateParameters(typeScriptSnippetFromSource(source.source, location, strictForExample), {
-            [SnippetParameters.$PROJECT_DIRECTORY]: directory,
-            ...source.metadata,
-          });
-          ret.push(fixturize(snippet, looseForExample));
-          break;
-        case 'markdown':
-          for (const snippet of extractTypescriptSnippetsFromMarkdown(source.markdown, source.location, strict)) {
-            const withDirectory = updateParameters(snippet, {
-              [SnippetParameters.$PROJECT_DIRECTORY]: directory,
-            });
-            ret.push(fixturize(withDirectory, loose));
-          }
-      }
-    }
-  }
-  return ret;
+export async function allTypeScriptSnippets(
+  assemblies: readonly LoadedAssembly[],
+  loose = false,
+): Promise<TypeScriptSnippet[]> {
+  return Promise.all(
+    assemblies
+      .flatMap((loaded) => allSnippetSources(loaded.assembly).map((source) => ({ source, loaded })))
```

Comment:
Oh jeez, I though `flatMap` was a node 14+ and realizing that it is, in fact, node 11+ kind of made my day 🙃